src/core/packages/saved-objects/migration-server-internal/src/run_v2_migration.ts

```diff
@@ -120,7 +163,10 @@ export const runV2Migration = async (options: RunV2MigrationOpts): Promise<Migra
   indicesWithRelocatingTypes.forEach((index) => migratorIndices.add(index));
 
   // we will store model versions instead of hashes (to be FIPS compliant)
-  const appVersions = getVirtualVersionMap(options.typeRegistry.getAllTypes());
+  const appVersions = getVirtualVersionMap({
```

Comment:
We store this Map (type=>modelVersion) in the meta information of the index.

With the `useModelVersionsOnly` flag we default to 10.0.0 for types that do NOT define modelVersions. This preserves the behavior that we had for all non-external SO types thanks to the [defaulting](https://github.com/elastic/kibana/blob/75c0f9543acc80e0d875e908f8c4570581393e20/src/core/packages/saved-objects/server-internal/src/apply_type_defaults.ts#L31) of the `switchToModelVersionAt`.